cEP-0024.md

```diff
@@ -0,0 +1,106 @@
+# Implement GitMate automations and plugins for coala
+
+| Metadata |                                                       |
+| -------- | ----------------------------------------------------- |
+| cEP      | 24                                                    |
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to the coala workflow and the necessary
+automation to realise it, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there are a number of missing features which would save time of
+our community, for example automatically close PRs from master. Many of them
+are general plugins, which may be useful to other any organisation, but coala
+needs more plugins which support coala development workflow. A team class in
+IGitt, representing GitHub and GitLab teams, will be implemented.
+
+## New Features
+
+### Auto-assign issue based on commit mention in a PR
+
+When a PR is opened in the repo and if any of it's commit message
+mentions (fixes, closes) any unassigned issues, auto assign commit author to the
+mentioned unassigned issues. Comment on the issue (configurable) if someone
+else is working on the mentioned issue.
```

Comment:
How do we know if someone else is working on the issue or not? The issue is unassigned, right?

Reply:
And in case of someone is assigned to the issue, why to comment on the issue that someone is working on it? I think its understandable.

But Yeah, if a user creates a pull request and someone else is assigned to the mentioned issue, GitMate should add a comment on the 'pull request' saying that the issue you mentioned in your commit is assigned to someone else and reject the PR.